core/src/services/aliyun_drive/core.rs

```diff
@@ -0,0 +1,506 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::Debug;
+use std::sync::Arc;
+
+use bytes::{Buf, Bytes};
+use chrono::Utc;
+use http::header::HeaderValue;
+use http::{Method, Request};
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use tokio::sync::Mutex;
+
+use crate::raw::*;
+use crate::*;
+
+use super::error::parse_error;
+
+/// Available Aliyun Drive Type.
+#[derive(Debug, Deserialize, Default, Clone)]
+pub enum DriveType {
+    /// Use the default type of Aliyun Drive.
+    #[default]
+    Default,
+    /// Use the backup type of Aliyun Drive.
+    ///
+    /// fallback to the default type if no backup drive found.
+    Backup,
+    /// Use the resource type of Aliyun Drive.
+    ///
+    /// fallback to the default type if no resource drive found.
+    Resource,
+}
+
+pub struct AliyunDriveTokener {
+    pub drive_id: Option<String>,
+    pub access_token: Option<String>,
+    pub refresh_token: String,
+    pub expire_at: i64,
+}
+
+pub struct AliyunDriveCore {
+    pub endpoint: String,
+    pub root: String,
+    pub client_id: String,
+    pub client_secret: String,
+    pub drive_type: DriveType,
+    pub rapid_upload: bool,
+
+    pub tokener: Arc<Mutex<AliyunDriveTokener>>,
+    pub client: HttpClient,
+}
+
+impl Debug for AliyunDriveCore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("AliyunDriveCore")
+            .field("root", &self.root)
+            .field("drive_type", &self.drive_type)
+            .finish_non_exhaustive()
+    }
+}
+
+impl AliyunDriveCore {
+    async fn send(&self, mut req: Request<Buffer>, token: Option<&str>) -> Result<Buffer> {
+        if req.method() == Method::POST {
+            req.headers_mut().insert(
+                "Content-Type",
```

Comment:
Please use `http::header::CONTENT_TYPE` which takes less memory and more safe to use.